clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -727,12 +733,22 @@ private void validateOffsetsAsync(Map<TopicPartition, SubscriptionState.FetchPos
                 return;
             }
 
-            subscriptions.setNextAllowedRetry(dataMap.keySet(), time.milliseconds() + requestTimeoutMs);
-
             final Map<TopicPartition, Metadata.LeaderAndEpoch> cachedLeaderAndEpochs = partitionsToValidate.entrySet()
                     .stream()
                     .collect(Collectors.toMap(Map.Entry::getKey, entry -> entry.getValue().currentLeader));
 
+            NodeApiVersions nodeApiVersions = apiVersions.get(node.idString());
+            if (nodeApiVersions == null || nodeApiVersions.latestUsableVersion(ApiKeys.OFFSET_FOR_LEADER_EPOCH) < 3) {
+                log.debug("Could not validate fetch offsets for partitions {} since the broker is not on " +
+                        "a late enough version (i.e. 2.3 and up)", cachedLeaderAndEpochs.keySet());
```

Comment:
Nit: `late enough version` reads a bit funny. Maybe we can say "Could not validate fetch offsets for partitions {} since the broker does not support the required protocol version (introduced in Kafka 2.3)" or something.